src/config/kubefile.rs

```diff
@@ -104,6 +107,91 @@ pub struct ContextConf {
     pub user: String,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+#[allow(non_snake_case)]
+pub struct Exec {
+    apiVersion: String,
+    pub args: Option<Vec<String>>,
+    pub command: Option<String>,
+    pub env: Option<Vec<Env>>,
```

Comment:
Can you not just make this an `Option<HashMap<String, String>>` directly?  I think serde should just "do the right thing" in that case.

Reply:
I tried this and the suggestions below. It compiles fine, but when I start click it gives the following error which I don't understand:

```
$ cargo build && ./target/debug/click
   Compiling click v0.4.2 (/Users/user/Code/click/click)
    Finished dev [unoptimized + debuginfo] target(s) in 8.77s
Could not load kubernetes config. Cannot continue.  Error was: Couldn't read yaml in '/Users/user/.kube/config': invalid type: sequence, expected a map
```
I'm not understanding why, as the generate_token function is not even called when starting, just after the first request to the API.

Any ways I can debug this faster?

Reply:
@nicklan 
Apparently serde doesn't like parsing the following when it's a 

`Option<HashMap<String, String>>`

```
users:
- name: eks-acceptance
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - eu-west-1
      - eks
      - get-token
      - --cluster-name
      - eks-acceptance
      command: aws
      env:
      - name: AWS_PROFILE
        value: default
```
I suppose it's something you could test too, it happens just after starting the application. For some reason it doesn't like the `env`. Not sure why.